src/components/QuestionPlay/WithoutOptions.tsx

```diff
@@ -18,14 +20,50 @@ export default function WithoutOptions({
   options,
   isAttempted,
   setIsAnswerRevealed,
-  setSelectedChoices,
+  submitResponse,
+  continueGame,
 }: Props) {
-  function handleSelectedChoice(choice: number[]) {
-    setSelectedChoices(choice);
+  const [isUndoTimerRunning, setIsUndoTimerRunning] = useState(false);
+  const [time, setTime] = useState(0);
+  const [isUndoClicked, setIsUndoClicked] = useState(false);
+  const timerRef = useRef(null as any);
+  const userResponse = useRef(null as any);
 
-    setTimeout(() => {
-      document.getElementById('btnSubmitResponse')?.click();
-    }, 100);
+  useEffect(() => {
+    if (isUndoTimerRunning) {
+      timerRef.current = setTimeout(() => {
+        if (time === 4) {
+          handleContinueClick(userResponse.current);
+        }
+
+        setTime(time + 1);
+      }, 1000);
+    }
+
+    return () => {
+      clearTimeout(timerRef.current);
+    };
+  }, [isUndoTimerRunning, time]);
+
+  function handleContinueClick(choice: number[]) {
+    submitResponse(choice);
+    continueGame();
+  }
+
+  function handleSubmitResponse(choice: number[]) {
+    if (isUndoClicked) {
+      submitResponse(choice);
```

Comment:
Submit response when undo button clicked? Feels odd.

Reply:
Is this `wasUndoClickedOnce`?

Reply:
yes.. changed the name